File: Labb1_task1/ReverseText.c
#include <stdio.h>
char* recursiveReverse(char* str, int len)
{
    if(len >= 0)
    {
        putchar(*str);
        str--;
        len--;
        recursiveReverse(str,len);
    }
    char* text = "\nyou chose recursive";
    return text;
}

void iterativeReverse(char* str, int len)
{
    int i = 0;
    for(i; i < len+1; i++)
    {
        putchar(*str);
        str--;
    }
}

int main(void)
{
    char ch;
    char* str;
    int length = 0;
    printf("enter your characters ");
    while(ch != '\n')
    {
        ch = getchar();
        *str = ch;
        str++;
        length++;
    }
    printf("type 1 for iterative, else recursive ");
    int i;
    scanf("%d", &i);
    if(i == 1)
    {
        iterativeReverse(str, length);
        printf("\nyou chose iterative \n");
    }
    else
       printf(recursiveReverse(str, length));
    
    return 0;
}

File: Labb1_task2/ReverseText.java
package Labb1_task2;

import java.util.Scanner;
import java.util.Stack;

public class ReverseText
{
    public static void main(String[] args)
    {
        Scanner sys = new Scanner(System.in);

        System.out.println("Enter your text ");

        String text = sys.nextLine();
        char[] textArr = new char[text.length()];
        for(int i = 0; i < text.length(); i++)
            textArr[i] = text.charAt(i);

        Stack stack = new Stack();
        for(int i = 0; i < textArr.length; i++)
            stack.push(textArr[i]);

        System.out.println("Type 1 for iterative, else recursive");
        int i = sys.nextInt();
        if(i == 1)
            System.out.println(iterativeReverse(stack));
        else
            recursiveReverse(stack);

    }

    public static String iterativeReverse(Stack stack)
    {
        StringBuilder sb = new StringBuilder("");
        while(!stack.isEmpty())
            sb.append(stack.pop());

        return sb.toString();
    }

    public static void recursiveReverse(Stack stack)
    {
        while(!stack.isEmpty())
        {
            System.out.print(stack.pop());
            recursiveReverse(stack);
        }

    }

}


